browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -215,26 +222,55 @@ class AdblockCnameResolveHostClient : public network::mojom::ResolveHostClient {
   }
 };
 
+// Called after checking a request without any CNAME uncloaking.
+void OnShouldBlockRequestResult(
+    scoped_refptr<base::SequencedTaskRunner> task_runner,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  if (ctx->blocked_by == kAdBlocked) {
+    brave_shields::DispatchBlockedEvent(
+        ctx->request_url, ctx->render_frame_id, ctx->render_process_id,
+        ctx->frame_tree_node_id, brave_shields::kAds);
+    next_callback.Run();
+  } else {
+    // If not blocked, the request still needs to be CNAME uncloaked.
+
+    // DoH or standard DNS queries won't be routed through Tor, so we need to
+    // skip it.
+    if (ctx->browser_context->IsTor() || ctx->did_match_important) {
+      next_callback.Run();
+    } else {
+      new AdblockCnameResolveHostClient(std::move(next_callback), task_runner,
```

Comment:
memory leak?

Reply:
Should be the same amount of memory leakage as before (i.e. hopefully none); I only moved that from https://github.com/brave/brave-core/pull/8704/files#diff-0986ef9872f15c706dc15f78ee8ef549348a10918b13d126e4483261d7171853L235-L236

FWIW it should get `delete`d at the end of `AdblockCnameResolveHostClient::OnComplete`:

https://github.com/brave/brave-core/blob/b4c864fc2345e52132363a8d00cecfbfbc08b286/browser/net/brave_ad_block_tp_network_delegate_helper.cc#L197-L212

Reply:
Got it. I would just add a comment. "This will delete itself upon completion"

Reply:
added a comment :smiley: